tests/common/test_setup_precheck.py

```diff
@@ -0,0 +1,157 @@
+"""Test included script to initialize and evaluate precheck permissions."""
+
+import contextlib
+import io
+import re
+import unittest
+from unittest.mock import Mock, patch
+
+import __init__
+
+import tests.common.test_common_constants as TestCommonConstants
+from pyracf import setup_precheck
+from pyracf.common.irrsmo00 import IRRSMO00
+
+# Resolves F401
+__init__
+
+
+@patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+class TestSetupPrecheck(unittest.TestCase):
```

Comment:
Theoretically yes, I understand where this makes sense, and would like to do it. Functionally, I cannot think of a case where this would be true. Our `SecurityRequestError` should not be raised by this script as it is currently defined. The Profile name meets the criteria for the XFACILIT class, and XFACILIT is a default class included with RACF. This also doesn't meet any known criteria for `DownstreamFatalError` as it is not running as a userid or performing any "alter" options that would require precheck. It is possible that it is capable of failure, but its surfacing of the error message is just a matter of python bringing that up as it has no special path of its own to raise such an exception, so I feel like this is functionally tested by our tests of `add_resource_profile`. I am open to further discussion though.

Reply:
That is a good point. We know that add resource can raise an exception and that has already technically been tested. However, I think it would be good to have a test just to demonstrate the exception raised from failing to add the resource gets manifested correctly.

Reply:
I think that should be sufficient. I think the key here is to demonstrate that if a `SecurityRequestError` is raised during the add, it gets manifested properly.

My only suggestion here is to change the name of the test function to `test_setup_precheck_throws_error_when_add_resource_fails`

Reply:
changed.